### PR TITLE
[12.0][FIX+IMP] rma, rma_sale: fix bugs and add improvements

### PR DESCRIPTION
**Pull request by ernestotejeda**
- Fix thrown error when trying to download a picking from the portal.
- Add the hook method to prepare RMA values ​​from the return pick wizard.
- Add the access rule for portal users.
- Show the portal 'Request RMAs' button on the sales page only to users
related to the sales order.

cc @Tecnativa TT25187

### CODE REVIEW

File: rma/models/rma_team.py [hunk 1]
```diff
@@ -8,9 +8,8 @@ class RmaTeam(models.Model):
     _name = "rma.team"
     _inherit = ['mail.alias.mixin', 'mail.thread']
     _description = "RMA Team"
-    _order = "sequence, name"
+    _order = "name"
 
-    sequence = fields.Integer()
```
Comment (chienandalu) [inline rma/models/rma_team.py]: Why to remove this?

Comment (ernestotejeda) [reply rma/models/rma_team.py]: I'm sorry, I added it again

**Review by chienandalu — APPROVED** [review]
Reviewed. Thanks!

File: rma/models/rma_team.py (hunk 1 above)
Comment (chienandalu) [reply rma/models/rma_team.py]: :smile: :+1:

File: rma_sale/views/sale_portal_template.xml [hunk 2]
```diff
@@ -1,7 +1,11 @@
 <odoo>
     <template id="sale_order_portal_template" name="Request RMA" inherit_id="sale.sale_order_portal_template">
         <xpath expr="//div[hasclass('o_portal_sale_sidebar')]//div[hasclass('o_download_pdf')]/.." position="after">
-            <li t-if="sale_order.state in ['sale', 'done']" class="list-group-item flex-grow-1" id="li-request-rma">
+            <!-- If the sale order was not linked to the partner of the logged user, the button will not be shown -->
+            <li t-if="sale_order.state in ['sale', 'done'] and env.user.partner_id.commercial_partner_id == sale_order.partner_id.commercial_partner_id"
```
Comment (pedrobaeza) [inline rma_sale/views/sale_portal_template.xml]: I don't think we should restrict this: if you are able to see the sales orders due to the permissions, then we can request an RMA.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
The portal record rule for RMAs is still missing

File: rma_sale/views/sale_portal_template.xml (hunk 2 above)
Comment (chienandalu) [reply rma_sale/views/sale_portal_template.xml]: Good point

**Comment by chienandalu** [comment]
> The portal record rule for RMAs is still missing

I see it here: https://github.com/OCA/rma/pull/161/files#diff-3f3c73a654f2fb27560b5b8a8834b6e0R44-R49

**Comment by pedrobaeza** [comment]
Oh, I don't know how I missed it. The other comment still applies though.

**Comment by ernestotejeda** [comment]
@pedrobaeza @chienandalu Changes done. Thanks.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**